Unix/samples/Providers/Indication/util/util.h

```diff
@@ -246,7 +254,11 @@ MI_INLINE void Sem_Destroy(
 #else
     if (self->sem)
     {
+#ifdef __APPLE__
+    dispatch_release(self->sem);
```

Comment:
please use pal/sem.h Sem structure and the associated semaphore implementation in there instead. That does all the platform specific stuff needed to implement semaphore on different platforms.